### DISCUSSION TITLE
How to set NumberBox background color to Transparent?

### DISCUSSION THREAD

**Discussion by greatoceansoftware** [microsoft/microsoft-ui-xaml#8325] [Q&A] [answered]
I've tried "lightweight styling" by setting Resources on the NumberBox control to change the background color of a NumberBox to Transparent. Nothing I've tried works. I've tried the following theme colors:

CardBackgroundFillColorDefaultBrush
LayerFillColorDefaultBrush
SolidBackgroundFillColorBaseBrush
ControlFillColorDefaultBrush

Any ideas?

            <NumberBox.Resources>
              <ResourceDictionary>
                <ResourceDictionary.ThemeDictionaries>
                  <ResourceDictionary x:Key="Light">
                    <SolidColorBrush x:Key="CardBackgroundFillColorDefaultBrush" Color="Transparent"/>
                    <SolidColorBrush x:Key="LayerFillColorDefaultBrush" Color="Transparent"/>
                    <SolidColorBrush x:Key="SolidBackgroundFillColorBaseBrush" Color="Transparent"/>
                    <SolidColorBrush x:Key="ControlFillColorDefaultBrush" Color="Transparent"/>
                  </ResourceDictionary>
                </ResourceDictionary.ThemeDictionaries>
              </ResourceDictionary>
            </NumberBox.Resources>

**Comment by castorix** [comment, marked as answer]
It depends on the state
For example for focus : _TextControlBackgroundFocused_

![WinUI3_NumberBox](https://user-images.githubusercontent.com/22345506/227787877-a749a929-d402-4827-acce-c9b044c78cca.gif)

**Reply by greatoceansoftware** [reply]
Thanks. This stuff is not well documented. The ones that achieve the effect I'm looking for are below.

The documentation for TextBox indicates that its resources affect other controls, but does not include NumberBox.

_Resources that start with TextControl are shared by TextBox, PasswordBox, RichEditBox, and AutoSuggestBox. Changes to these resources will affect all four controls._


            <NumberBox.Resources>
              <ResourceDictionary>
                <ResourceDictionary.ThemeDictionaries>
                  <ResourceDictionary x:Key="Light">
                    <SolidColorBrush x:Key="TextControlBackground" Color="Transparent"/>
                    <SolidColorBrush x:Key="TextControlBorderBrushFocused" Color="Transparent"/>
                  </ResourceDictionary>
                </ResourceDictionary.ThemeDictionaries>
              </ResourceDictionary>
            </NumberBox.Resources>
